### PR TITLE
Implement tasks from Module 5

### PR DESCRIPTION
**Pull request by f8B23Lef**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "lint": "eslint -c .eslintrc.json ./src/**/*.js",
     "fix-lint": "eslint -c .eslintrc.json ./src/**/*.js --fix",
     "kill-node": "taskkill /F /IM node.exe",
-    "task3": "nodemon --exec babel-node ./src/app.js"
+    "task4": "nodemon --exec babel-node ./src/app.js"
   },
```
Comment (ghost) [inline package.json:12]: please add `.vscode` to gitignore to ignore vscode settings

File: src/services/user.js [hunk 2]
```diff
@@ -35,11 +37,19 @@ export default class UserService {
...
   static async deleteUser(id) {
     const isUserExist = await findUserById(id);
-    if (isUserExist) {
-      await deleteUser(id);
-    } else {
+    if (!isUserExist) {
       throw new NotFoundError(`User with id = ${id} cannot be deleted`);
     }
+
+    const transaction = await sequelize.transaction();
```
Comment (ghost) [inline src/services/user.js]: It is better to deal with transactions only on data-acess layer. Transactions are not a part of business logic, but service should contain business logic only without coupling it to specific storage type an so on. 

Lets imagine that in the future we will use another kind of a storage where we wont need transactions. In this case our service layer should stay unchanged and we will just replace data-acess layer implementation

File: src/services/user.js [hunk 3]
```diff
@@ -35,11 +37,19 @@ export default class UserService {
...
+    try {
+      await deleteUser(id, transaction);
+      await deleteUserFromGroups(id, transaction);
```
Comment (ghost) [inline src/services/user.js]: By the way, if I'm not mistaken postgres can automatically delete all connected to user rows when user is deleted. Please look into `CASCADE` delete

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/services/user.js (hunk 3 above)
Comment (f8B23Lef) [reply src/services/user.js]: Yes, but we don't delete user from Users table. We just mark it `isDeleted: true`. So we should remove users from UserGroup table manually.

File: src/services/user.js (hunk 2 above)
Comment (f8B23Lef) [reply src/services/user.js]: Done

File: package.json (hunk 1 above)
Comment (f8B23Lef) [reply package.json:12]: Added

**Review by ghost — APPROVED** [review]

**Pull request merged**